File: 2094-remove-stones-to-minimize-the-total/2094-remove-stones-to-minimize-the-total.py
class Solution:
    def minStoneSum(self, piles: List[int], k: int) -> int:
        heap=[]
        for i in piles:
            heapq.heappush(heap,-i)
        for i in range(k):
            largest=-heapq.heappop(heap)
            heapq.heappush(heap,-largest//2)
        return -sum(heap)
        

File: 2094-remove-stones-to-minimize-the-total/2094-remove-stones-to-minimize-the-total.java
class Solution {
    public int minStoneSum(int[] piles, int k) {
        PriorityQueue<Integer> pq=new PriorityQueue<>();
        for(int i:piles){
            pq.add(-i);
        }
        int sum=0;
        for(int i=0;i<k;i++){
            int largest=-pq.poll();
            pq.add(-(largest-largest/2));
        }
        while(!pq.isEmpty()){
            sum+=-pq.poll();
        }
        return sum;
    }
}